### PR TITLE
[SPARK-14772][ML,PySpark]Python ML Params.copy treats uid, paramMaps …

### PR DESCRIPTION
**Pull request by hujy**
## What changes were proposed in this pull request?

The patch referenced [SPARK-14772](https://issues.apache.org/jira/browse/SPARK-14772#)
After apply this patch, user can choose to just copy the default param map or the param map.
## How was this patch tested?

Unit test, test locally.

### CODE REVIEW

File: python/pyspark/ml/param/__init__.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def getOrDefault(self, param):
             return self._defaultParamMap[param]
 
     @since("1.4.0")
-    def extractParamMap(self, extra=None):
```
Comment (holdenk) [inline python/pyspark/ml/param/__init__.py]: Please document this new param

**Comment by holdenk** [comment]
Thanks for tackling this issue :) For a better understanding -  is there a reason why adding a flag for this behaviour instead of just changing it (since it is a bug) - do we expect people to want to explicitly copy the default param map in this way?

**Comment by holdenk** [comment]
Also it would be good to have tests to ensure the change has the desired impact.

**Comment by hujy** [comment]
@holdenk, I'm checking with @jkbradley about  "respect the difference between defaultParamMap and paramMap" in the requirement. The new input parameter is optional and will not impact the user not want to copy default param.  : )

**Comment by hujy** [comment]
ok with test

**Comment by holdenk** [comment]
So to trigger the test we will need one of the committers - e.g. @davies is one of the more active Python committers (although he has been busy lately) so we can also check with @MLnick . It would also be nice if you could document the param (even if we don't expect the user to use it - its useful to have it noted for other developers and so users know not to worry about it).

**Comment by jkbradley** [comment]
@hujy Thank you for sending this PR, and apologies for not seeing it earlier.  Since the other PR for this JIRA is ready to merge, could you please close this issue?  Thanks again!

**Pull request closed without merging**